helper/helper.go

```diff
@@ -0,0 +1,81 @@
+package helper
+
+import (
+	"archive/tar"
+	"errors"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+func GetApplication(applicationsPath, applicationUUID string) (string, string, error) {
+	commit, err := getCommit(applicationsPath, applicationUUID)
+	if err != nil {
+		log.Printf("Failed to get commit. Error: %s\r\n", err)
+		return "", "", err
+	}
+
+	basePath := filepath.Join(applicationsPath, applicationUUID, commit)
+	tarball := filepath.Join(basePath, "binary.tar")
+	application := filepath.Join(basePath, "application.zip")
+
+	_, err = os.Stat(application)
+	if err != nil {
+		err = untar(tarball, basePath)
+		if err != nil {
+			log.Printf("Failed to extract application. Error: %s\r\n", err)
+			return "", "", err
+		}
+	}
+
+	return application, commit, err
+}
+
+func untar(tarball, target string) error {
+	reader, err := os.Open(tarball)
```

Comment:
This isn't returning a reader, it's returning a `*File`. It's confusing because the Reader type (see [io package definition](https://golang.org/pkg/io/#Reader)) does not actually implement a `Close()` method.